internal/acceptance/openstack/loadbalancer/v2/loadbalancer.go

```diff
@@ -680,3 +682,72 @@ func WaitForLoadBalancerState(client *gophercloud.ServiceClient, lbID, status st
 		return false, nil
 	})
 }
+
+func CreateFlavorProfile(t *testing.T, client *gophercloud.ServiceClient) (*flavorprofiles.FlavorProfile, error) {
+	flavorProfileName := tools.RandomString("TESTACCT-", 8)
+	flavorProfileDriver := "amphora"
+	flavorProfileData := "{\"loadbalancer_topology\": \"SINGLE\"}"
+
+	createOpts := flavorprofiles.CreateOpts{
+		Name:         flavorProfileName,
+		ProviderName: flavorProfileDriver,
+		FlavorData:   flavorProfileData,
+	}
+
+	flavorProfile, err := flavorprofiles.Create(client, createOpts).Extract()
+	if err != nil {
+		return flavorProfile, err
+	}
+
+	t.Logf("Successfully created flavorprofile %s", flavorProfileName)
+
+	th.AssertEquals(t, flavorProfile.Name, flavorProfileName)
```

Comment:
The order of the args for these asserts should be reversed.
It should be `expected, actual`. This helps debugging issues, as these values are printed when they don't match.